src/OpenTelemetry/Sdk.cs

```diff
@@ -51,9 +55,41 @@ static Sdk()
         /// <param name="textMapPropagator">TextMapPropagator to be set as default.</param>
         public static void SetDefaultTextMapPropagator(TextMapPropagator textMapPropagator)
         {
+            Guard.ThrowIfNull(textMapPropagator);
```

Comment:
I suspect we have other places this question applies, but for public APIs should we notate possible exceptions in the comments?

```csharp
/// <exception cref="NullReferenceException">
/// Thrown when the <c>textMapPropagator</c> is null.
/// </exception>
```

Reply:
I think this would be nice to have, but we would need some tooling to identify.
- would need to discover them all
- would need to keep it up to date when exceptions are added or removed

Reply:
Totally agree with this. We _should_ have these comments but without tooling support I worry it will often be wrong, missing, or outright lies 😄

Reply:
Are you saying you're trying to replace good jobs with machines? This PR got political all of the sudden 😆 

Jokes aside, yes I completely agree and it's a discussion that's beyond the scope of this PR.

Reply:
> Are you saying you're trying to replace good jobs with machines? This PR got political all of the sudden 😆

Machines probably don't need such `<exception>` comment, I assume they will just disassemble the assembly to figure out what exception can be thrown from the callee. 🤣